### ISSUE TITLE
`no-duplicate-selectors` in Sass dynamic selectors

### ISSUE THREAD

**Issue by julien-deramond** [stylelint/stylelint#7891]
### What minimal example or steps are needed to reproduce the bug?

```scss
$var-test: "x" !default;

.a#{$var-test}-a {
  &.b {
    color: #fd0;
  }
}

.a#{$var-test}-b {
  &.b {
    color: #fd0;
  }
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "no-duplicate-selectors": true
  }
}
```

### How did you run Stylelint?

[Stylelint demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIAkA3AhgJwLQBcYGcd4ACAHRAA9ziBCWAMzQFcAbHAblIDtuA6NAMTBUmXARwBfLGmLBuxYgDJeAI1nyFxSCwgYSA+lAAMnLgoncLPLvyEjseQlLVyzS1ercLtu-YZMaVlYgADQg9ACWLDAAcmgAtnCIMBQJAA7RvPhg+Pih4BBckQDmCCCuCuQpeFxQ+OQkANoalSCEAJ7RLBFcOFiQRRHFWBgwkPGJtTBQWNm55BoAupbc+QMlAGK68Wg4ZQBW+IX5sGl5iBVkbWjzCFcsu+LkIRrkHV09OA33j4TPr20cJ0YN1ev1CiURmMIBMYFMZnN6ndyA9HF8QCsuCAJEA).

### Which Stylelint-related dependencies are you using?

```json
{
  "sass": "latest",
  "stylelint": "latest",
  "stylelint-config-recommended-scss": "latest"
}
```

### What did you expect to happen?

This Sass code will generate the following CSS code:

```css
.ax-a.b {
  color: #fd0;
}

.ax-b.b {
  color: #fd0;
}
```

The generated selectors are not duplicate and different, there shouldn't be any warnings for the corresponding Sass code.
It wasn't the case in 16.7.0.

### What actually happened?

In the provided demo example, the following error/warning is raised:

```
Unexpected duplicate selector "&.b", first used at line 4
[(no-duplicate-selectors)](https://stylelint.io/user-guide/rules/no-duplicate-selectors)
[10:3-6]
```

It wasn't the case in 16.7.0.

More context in an actual use case at https://github.com/twbs/bootstrap/pull/40676#issuecomment-2257575624

### Do you have a proposal to fix the bug?

I'm not familiar with the stylelint codebase, but if needed, I could try to find some time to help to fix this issue. Please let me know.

**Comment by ybiquitous** [comment]
@julien-deramond Thanks for the report with a reproducible demo. This seems to be the same issue as #7887, so can you confirm with the edge version?

```json
{
  "dependencies": {
    "stylelint": "stylelint/stylelint"
  }
}
```

Or can you please wait for the next version? I'll release it today or tomorrow.

**Comment by julien-deramond** [comment]
Thank you @ybiquitous, I wasn't sure it was exactly the same issue. I've tried the edge version, and the warnings/errors disappear. So I'm closing this issue, we'll wait for the next version :)
